app/services/set_decline_by_default.rb

```diff
@@ -0,0 +1,38 @@
+class SetDeclineByDefault
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call
+    return false if pending_application_choices.any? || application_choices.none?(&:offered_at)
+
+    most_recent_decision_date = [
+      application_choices.maximum(:offered_at),
+      application_choices.maximum(:rejected_at),
+    ].compact.max
+
+    dbd_days = TimeLimitCalculator.new(
+      rule: :decline_by_default,
+      effective_date: most_recent_decision_date,
+    ).call
+
+    dbd_date = dbd_days.business_days.after(most_recent_decision_date).end_of_day
+
```

Comment:
Should we split the `offer` state in `application_choice` into 2: `offer_made_but_not_sent` and `offer_sent`? Would that make this easier to build?

Reply:
Depends if these should be treated as independent states. My preference would be to avoid this, it feels like we abusing states for book-keeping purposes, such as knowing if an email has been sent. Unless you mean something different by ```not_sent``` and ```sent``` (e.g. candidate accepting offer while other provider decisions are still pending -- yes this is thing -- but it should be fine with the new checks).

Reply:
I think I agree with Michael, but I would like to check my understanding.

AFAICS the way this works at the moment is that the checks at the top of this procedure mean it only runs once all the choices are ready. In that sense it's ready to become a more generic place, as @tijmenb suggests above in his comment about renaming, to hook actions that need to happen once the batch as a whole is decided on, like emailing the candidate.

I think there's another thing to check here though, which is what happens when this gets the same application_form twice? if the dbd date is already set, it shouldn't change it.